common/types/types.ts

```diff
@@ -62,3 +63,19 @@ export type TopicIds = {
   topicId?: string;
   subTopicId?: string;
 };
+
+export type Data =
+  | {
+      topics: Topic[];
+      languages: Language[];
+      translations: Translations;
+    }
+  | undefined;
```

Comment:
I don't think we should add `undefined` to the type's union, but rather add it whenever we're using the type.

```suggestion
export type Data =
  | {
      topics: Topic[];
      languages: Language[];
      translations: Translations;
    };
```